### PR TITLE
ci: fix potential GitHub Actions smells

### PR DESCRIPTION
**Pull request by ceddy4395**
### 📚 Description

Hey! 🙂
I want to contribute the following changes to your workflow:

- Use 'if' for upload-artifact action
- Avoid uploading artifacts on forks
- Avoid executing  scheduled workflows on forks

(These changes are part of a research Study at TU Delft looking at GitHub Action Smells. [Find out more](https://ceddy4395.github.io/research/gha-smells.html))

### CODE REVIEW

File: .github/workflows/reproduire-close.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ permissions:
 jobs:
   stale:
     runs-on: ubuntu-latest
+    if: ${{github.event_name == 'workflow_dispatch' || github.repository == 'nuxt/nuxt.js'}}
```
Comment (danielroe) [inline .github/workflows/reproduire-close.yml:13]:
```suggestion
    if: ${{github.event_name == 'workflow_dispatch' || github.repository == 'nuxt/nuxt'}}
```

File: .github/workflows/scorecards.yml [hunk 2]
```diff
@@ -28,6 +28,7 @@ jobs:
       id-token: write
       contents: read
       actions: read
+    if: ${{github.event_name == 'push' || github.repository == 'nuxt/nuxt.js'}}
```
Comment (danielroe) [inline .github/workflows/scorecards.yml:31]:
```suggestion
    if: ${{github.event_name == 'push' || github.repository == 'nuxt/nuxt'}}
```

File: .github/workflows/scorecards.yml [hunk 3]
```diff
@@ -58,14 +59,16 @@ jobs:
       # Upload the results as artifacts (optional). Commenting out will disable uploads of run results in SARIF
       # format to the repository Actions tab.
       - name: "Upload artifact"
-        uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.3
+        uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.1
+        if: ${{github.repository == 'nuxt/nuxt.js' && success()}}
```
Comment (danielroe) [inline .github/workflows/scorecards.yml:63]:
```suggestion
        if: ${{github.repository == 'nuxt/nuxt' && success()}}
```

File: .github/workflows/scorecards.yml [hunk 4]
```diff
@@ -58,14 +59,16 @@ jobs:
...
         with:
           name: SARIF file
           path: results.sarif
           retention-days: 5
 
       # Upload the results to GitHub's code scanning dashboard.
       - name: "Upload to code-scanning"
-        uses: github/codeql-action/upload-sarif@8f596b4ae3cb3c588a5c46780b86dd53fef16c52 # v3.25.2
+        uses: github/codeql-action/upload-sarif@8f596b4ae3cb3c588a5c46780b86dd53fef16c52 # v3.24.9
+        if: ${{github.repository == 'nuxt/nuxt.js' && success()}}
```
Comment (danielroe) [inline .github/workflows/scorecards.yml:72]:
```suggestion
        if: ${{github.repository == 'nuxt/nuxt' && success()}}
```

File: .github/workflows/scorecards.yml [hunk 5]
```diff
@@ -58,14 +59,16 @@ jobs:
       # Upload the results as artifacts (optional). Commenting out will disable uploads of run results in SARIF
       # format to the repository Actions tab.
       - name: "Upload artifact"
-        uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.3
+        uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.1
```
Comment (danielroe) [inline .github/workflows/scorecards.yml:62]: Is there a reason you change the version number?

File: .github/workflows/scorecards.yml [hunk 6]
```diff
@@ -58,14 +59,16 @@ jobs:
...
+        if: ${{github.repository == 'nuxt/nuxt.js' && success()}}
         with:
           name: SARIF file
           path: results.sarif
           retention-days: 5
 
       # Upload the results to GitHub's code scanning dashboard.
       - name: "Upload to code-scanning"
-        uses: github/codeql-action/upload-sarif@8f596b4ae3cb3c588a5c46780b86dd53fef16c52 # v3.25.2
+        uses: github/codeql-action/upload-sarif@8f596b4ae3cb3c588a5c46780b86dd53fef16c52 # v3.24.9
```
Comment (danielroe) [inline .github/workflows/scorecards.yml:71]: As above, is there a reason the comment is changed?

**Review by danielroe — COMMENTED** [review]
Thank you ❤️

File: .github/workflows/scorecards.yml (hunk 5 above)
Comment (ceddy4395) [reply .github/workflows/scorecards.yml]: I think this was a mistake during rebasing, my appologies!

File: .github/workflows/scorecards.yml (hunk 6 above)
Comment (ceddy4395) [reply .github/workflows/scorecards.yml]: Same here!

File: .github/workflows/reproduire-close.yml (hunk 1 above)
Comment (ceddy4395) [reply .github/workflows/reproduire-close.yml]: Ah I see I accidentally used the old repository name...

**Review by danielroe — APPROVED** [review]

**Pull request merged**